graviton/abi_strategy.py

```diff
@@ -3,17 +3,31 @@
 
 TODO: Leverage Hypothesis!
 """
+from abc import ABC, abstractmethod
 from collections import OrderedDict
 import random
 import string
-from typing import Callable, Dict, List, Optional, Union, cast
+from typing import Callable, Dict, List, Optional, Sequence, Union, cast
 
 from algosdk import abi, encoding
 
-PY_TYPES = Union[bool, int, list, str, bytes]
+PY_TYPES = Union[bool, int, Sequence, str, bytes]
 
 
-class ABIStrategy:
+class ABIStrategy(ABC):
+    @abstractmethod
+    def __init__(self, abi_instance: abi.ABIType, dynamic_length: Optional[int] = None):
```

Comment:
@tzaffi The highlighted parameters are unused.  Is it safe to remove them?  Or possibly the intent is to force usage with a `super` invocation?



```suggestion
    def __init__(self):
```

Reply:
👍 - Iooks like I can dispense with the abstract constructor in its entirety.